lib/ansible/modules/network/edgeos/edgeos_config.py

```diff
@@ -206,67 +242,118 @@ def check_command(module, command):
+    if delete_commands:
+        for line in set_commands:
+            search = re.sub('^set ', 'delete ', line)
+            for dline in delete_commands:
+                if search.startswith(dline):
+                    updates.append(line)
+
+    # Unmanaged config (config without matching commands)
+    unmanaged_config = (list(set(config) - set(set_commands)))
+    matches = list()
+    # Remove if actually a change to config
+    for line in unmanaged_config:
+        search = line.rsplit(' ', 1)[0]
+        for update in updates:
+            if update.startswith(search):
+                matches.append(line)
+                break
+
+    unmanaged_config = [line for line in unmanaged_config if line not in matches]
+
+    return updates, unmanaged_config, invalid_commands
 
-            if item not in config:
-                updates.append(line)
 
-            # If there is a corresponding delete command in the desired config, make sure to append
-            # the set command even though it already exists in the running config
-            else:
-                ditem = re.sub('set', 'delete', item)
-                for line in delete_commands:
-                    if ditem.startswith(line):
-                        updates.append(item)
-
-        elif item.startswith('delete'):
-            if not config:
-                updates.append(line)
-            else:
-                item = re.sub(r'delete', 'set', item)
-                for entry in config:
-                    if entry.startswith(item) and line not in visited:
-                        updates.append(line)
-                        visited.add(line)
+def run(module, result):
+    """compares config against passed configuration to asnible to create an
+    update list and applies to the edgeos device.
 
-    return list(updates)
+    .. warning:: docstring added long after code written, requires verification
+    of Arguments and Returns - please update if you see any errors
 
+    :param module: ansible module for self ref
+    :type module: ansible.module
+    :param result: result dict to be populated
+    process
+    :type result: dict
+    """
 
-def run(module, result):
     # get the current active config from the node or passed in via
     # the config param
     config = module.params['config'] or get_config(module)
 
     # create the candidate config object from the arguments
     candidate = get_candidate(module)
 
-    # create loadable config that includes only the configuration updates
-    commands = diff_config(module, candidate, config)
+    # create loadable config from updates, also return unmanaged and invalid
+    # commands
+    updates, unmanaged_config, invalid_commands = diff_config(module, candidate, config)
 
-    result['commands'] = commands
+    result['commands'] = sorted(updates)
+    result['unmanaged'] = sorted(unmanaged_config)
+    result['invalid'] = sorted(invalid_commands)
```

Comment:
Will sorting this output be confusing since it doesn't reflect the order in which the commands were sent to the device?

Reply:
oops, good point. Just making my life easier, dev first.

I have left the others as readability is improved through sorting